src/session/scout.c

```diff
@@ -140,14 +141,9 @@ _z_hello_list_t *_z_scout_inner(const z_what_t what, _z_id_t zid, const char *lo
     _z_scouting_message_encode(&wbf, &scout);
 
     // Scout on multicast
-#if Z_FEATURE_MULTICAST_TRANSPORT == 1
-    ret = __z_scout_loop(&wbf, locator, timeout, exit_on_first);
-#else
-    (void)(locator);
-    (void)(timeout);
-    (void)(exit_on_first);
-#endif  // Z_FEATURE_MULTICAST_TRANSPORT == 1
-
+    if (_zp_is_multicast_here()) {
```

Comment:
@Mallets this is the only place where the `is_multicast_here` function is used

Reply:
Looking at the code, it seems that scouting is only performed when *multicast transport* is activated.
However, https://github.com/jean-roland/zenoh-pico/blob/20640afe4d8f442272597b56604e514f84a0bb75/src/session/scout.c#L36 already provides the necessary check on UDP multicast support. Therefore, I believe that `_zp_is_multicast_here()` is redundant and could be removed.